tests/test_mozilla_reps.py

```diff
@@ -0,0 +1,250 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Santiago Dueñas <sduenas@bitergia.com>
+#     Alvaro del Castillo <acs@bitergia.com>
+#
+
+import argparse
+import json
+import shutil
+import sys
+import tempfile
+import unittest
+
+import httpretty
+
+if not '..' in sys.path:
+    sys.path.insert(0, '..')
+
+from perceval.cache import Cache
+from perceval.errors import CacheError
+from perceval.backends.mozilla_reps import MozillaReps, MozillaRepsCommand, MozillaRepsClient
+
+
+MOZILLA_REPS_SERVER_URL = 'http://example.com'
+MOZILLA_REPS_API = MOZILLA_REPS_SERVER_URL + '/api/v1/'
+MOZILLA_REPS_API_EVENTS = MOZILLA_REPS_SERVER_URL + '/api/v1/event/'
+
+
+def read_file(filename, mode='r'):
+    with open(filename, mode) as f:
+        content = f.read()
+    return content
+
+
+class HTTPServer():
+
+    requests_http = []  # requests done to the server
+
+    @classmethod
+    def routes(cls, empty=False):
+        """Configure in http the routes to be served"""
+
+        mozilla_events = read_file('data/mozilla_reps_events.json')
+        if empty:
+            mozilla_events = read_file('data/mozilla_reps_events_empty.json')
+
+        def request_callback(method, uri, headers):
+            body = mozilla_events
+
+            HTTPServer.requests_http.append(httpretty.last_request())
+
+            return (200, headers, body)
+
+        httpretty.register_uri(httpretty.GET,
+                               MOZILLA_REPS_API_EVENTS,
+                               responses=[
+                                    httpretty.Response(body=request_callback)
+                               ])
+
+    @classmethod
+    def check_events_contents(cls, testObj, events):
+        testObj.assertEqual(events[0]['data']['estimated_attendance'], 1000)
+        testObj.assertEqual(events[0]['data']['local_start'], '2012-06-06T10:00:00')
+        testObj.assertEqual(events[0]['data']['start'], '2012-06-06T02:00:00')
+        testObj.assertEqual(events[0]['origin'], MOZILLA_REPS_SERVER_URL)
+        testObj.assertEqual(events[0]['uuid'], 'e701d4ed3b954361383d678d2168a44307d7ff60')
+        testObj.assertEqual(events[0]['updated_on'], 1339326000.0)
+
+        if len(events) > 1:
+            testObj.assertEqual(events[1]['data']['estimated_attendance'], 50)
+            testObj.assertEqual(events[1]['data']['local_start'], '2012-06-09T09:00:00')
+            testObj.assertEqual(events[1]['data']['start'], '2012-06-09T07:00:00')
+            testObj.assertEqual(events[1]['origin'], MOZILLA_REPS_SERVER_URL)
+            testObj.assertEqual(events[1]['uuid'], 'e83e53b1f1039f62c66d2a977e23df3adf5bdc87')
+            testObj.assertEqual(events[1]['updated_on'], 1339344000.0)
+
+
+class TestMozillaRepsBackend(unittest.TestCase):
```

Comment:
It is not raise in the backend. The data received is plain JSON directly, so the parser is the json parser directly, we don't have any parsers to be tested in our code. Are you thinking in receiving bad JSON from the server for a network problem or something like that?